packages/neon/neon/lib/src/widgets/cached_image.dart

```diff
@@ -2,18 +2,81 @@ part of '../../neon.dart';
 
 final _cacheManager = DefaultCacheManager();
 
-abstract class NeonCachedImage extends StatefulWidget {
-  const NeonCachedImage({
+typedef APIImageDownloader = Future<Uint8List> Function();
+
+class NeonCachedImage extends StatefulWidget {
+  const NeonCachedImage._({
     required this.getImageFile,
+    required Key super.key,
     this.isSvgHint = false,
     this.size,
     this.fit,
     this.svgColor,
     this.iconColor,
-    super.key,
   });
 
-  final Future<File> Function() getImageFile;
+  factory NeonCachedImage.url({
+    required final String url,
+    final Size? size,
+    final BoxFit? fit,
+    final Color? svgColor,
+    final Color? iconColor,
+    final Key? key,
+  }) =>
+      NeonCachedImage._(
+        getImageFile: () async {
+          final file = await _cacheManager.getSingleFile(url);
+          return file.readAsBytes();
+        },
+        isSvgHint: Uri.parse(url).path.endsWith('.svg'),
+        size: size,
+        fit: fit,
+        svgColor: svgColor,
+        iconColor: iconColor,
+        key: key ?? Key(url),
+      );
+
+  factory NeonCachedImage.api({
+    required final Account account,
+    required final String cacheKey,
+    required final APIImageDownloader download,
```

Comment:
I wonder if it would be possible to have a hint based on the returned content type. Currently we don't have access to the raw API response so we can't access it, but maybe you could leave a todo note about that here.

Reply:
I have local changes with a FilesCachedImage where it just takes the FilesDetails and does exactly that. `details.mimeType?.contains('svg')` as a Hint.


And that's the reason why we should think of how we can extend this.